pkg/server/stream/v3/stream.go

```diff
@@ -23,10 +23,33 @@ type DeltaStream interface {
 
 // StreamState will keep track of resource state on a stream
 type StreamState struct {
-	// Indicates whether the original DeltaRequest was a wildcard LDS/RDS request.
-	IsWildcard bool
+	Wildcard bool
 
 	// ResourceVersions contains a hash of the resource as the value and the resource name as the key.
 	// This field stores the last state sent to the client.
 	ResourceVersions map[string]string
 }
+
+// NewStreamState initializes a stream state.
+func NewStreamState() StreamState {
+	return StreamState{
+		ResourceVersions: make(map[string]string),
+	}
+}
+
+// GetResourceVersions returns a new map snapshotted from the current state of the s.ResourceVersions map.
+// We do this because maps are reference types in Go, and if we pass around the ResourceVersions map into the cache
+// the state can be mutated across multiple threads creating a datarace.
+func (s StreamState) GetResourceVersions() map[string]string {
+	return s.ResourceVersions
+}
+
+// SetResourceVersions sets a new list of ResourceVersions in the StreamState
+func (s StreamState) SetResourceVersions(versions map[string]string) {
+	s.ResourceVersions = versions
+}
```

Comment:
with ResourceVersions being exposed I'm not sure if these helpers provide any value?

Reply:
Yup removed